// bawabah/src/index.ts
interface Credentials {
  appId: string;
  appSecret: string;
}

interface SessionRequestBody {
  appId: Credentials['appId'];
  appSecret: Credentials['appSecret'];
  sessionId: string;
}

interface CaptureResponse {
  openId: string;
  displayname: string | null;
  avatar: string | null;
  method: {
    type: "email" | "whatsapp";
    value: string;
  };
}

class Bawabah {
  private credentials: Credentials | null = null;

  private formatError(message: string): string {
    return `\x1b[31mBawabah Error:\x1b[37m ${message}\x1b[0m`;
  }

  init({ appId, appSecret }: Credentials): void {
    if (!appId && !appSecret) {
      throw new Error(this.formatError('Missing appId and appSecret. appId and appSecret is Missing in bw.init().'));
    }
    if (!appId) {
      throw new Error(this.formatError('Missing appId. appId is Missing in bw.init().'));
    }
    if (!appSecret) {
      throw new Error(this.formatError('Missing appSecret. appSecret is Missing in bw.init().'));
    }
    this.credentials = { appId, appSecret };
  }

  async capture(sessionId: string): Promise<CaptureResponse> {
    if (!this.credentials) {
      throw new Error(this.formatError('Library not initialized. Please call bw.init({appId: "your-app-id", appSecret: "your-app-secret"}) first.'));
    }

    if (!sessionId) {
      throw new Error(this.formatError('Missing sessionId. Please provide a sessionId to Capture.'));
    }

    const body: SessionRequestBody = {
      appId: this.credentials.appId,
      appSecret: this.credentials.appSecret,
      sessionId,
    };

    const response = await fetch('https://api.bawabah.app/v1/app/session', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(body),
    });

    if (!response.ok) {
      const errorData = await response.json();
      throw new Error(this.formatError(`${response.status} - ${errorData.error}`));
    }

    return response.json();
  }
}

export default Bawabah;
