// bawabah/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  default: () => index_default
});
module.exports = __toCommonJS(index_exports);
var Bawabah = class {
  constructor() {
    this.credentials = null;
  }
  formatError(message) {
    return `\x1B[31mBawabah Error:\x1B[37m ${message}\x1B[0m`;
  }
  init({ appId, appSecret }) {
    if (!appId && !appSecret) {
      throw new Error(this.formatError("Missing appId and appSecret. appId and appSecret is Missing in bw.init()."));
    }
    if (!appId) {
      throw new Error(this.formatError("Missing appId. appId is Missing in bw.init()."));
    }
    if (!appSecret) {
      throw new Error(this.formatError("Missing appSecret. appSecret is Missing in bw.init()."));
    }
    this.credentials = { appId, appSecret };
  }
  async capture(sessionId) {
    if (!this.credentials) {
      throw new Error(this.formatError('Library not initialized. Please call bw.init({appId: "your-app-id", appSecret: "your-app-secret"}) first.'));
    }
    if (!sessionId) {
      throw new Error(this.formatError("Missing sessionId. Please provide a sessionId to Capture."));
    }
    const body = {
      appId: this.credentials.appId,
      appSecret: this.credentials.appSecret,
      sessionId
    };
    const response = await fetch("https://api.bawabah.app/v1/app/session", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(body)
    });
    if (!response.ok) {
      const errorData = await response.json();
      throw new Error(this.formatError(`${response.status} - ${errorData.error}`));
    }
    return response.json();
  }
};
var index_default = Bawabah;
